### PR TITLE
Refactor GitHub services

### PR DESCRIPTION
**Pull request by takecare**
#### Problem

As described in #3 we have a bunch of service classes that are essentially 1-1 mappings to the API interface, not adding any valuable logic whatsoever; they were just an unnecessary middle layer.

#### Solution

Remove those classes and pass in the service API interface as a dependency directly.

##### Test(s) added

No, not yet.

### CODE REVIEW

File: batch-aws-lambda/src/main/java/com/novoda/github/reports/lambda/issue/CommentsServiceClient.java [hunk 1]
```diff
@@ -6,54 +6,57 @@
...
     private static final int DEFAULT_PER_PAGE_COUNT = 100;
 
-    private final IssueService issueService;
+    private final GithubApiService apiService;
     private final DateToISO8601Converter dateConverter;
     private final ResponsePersistTransformer<RepositoryIssueEvent> responseRepositoryIssueEventPersistTransformer;
 
     public static CommentsServiceClient newInstance() {
-        IssueService issueService = GithubIssueService.newInstance();
+        GithubApiService apiService = GithubCachingServiceContainer.getGithubService();
```
Comment (takecare) [inline batch-aws-lambda/src/main/java/com/novoda/github/reports/lambda/issue/CommentsServiceClient.java]: this should be `GithubServiceContainer.getGithubService()`

File: batch-aws-lambda/src/main/java/com/novoda/github/reports/lambda/issue/IssuesServiceClient.java [hunk 2]
```diff
@@ -6,55 +6,53 @@
...
     private static final GithubIssue.State DEFAULT_STATE = GithubIssue.State.ALL;
 
-    private final IssueService issueService;
+    private final GithubApiService apiService;
     private final DateToISO8601Converter dateConverter;
     private final ResponsePersistTransformer<RepositoryIssue> responseRepositoryIssuePersistTransformer;
 
     public static IssuesServiceClient newInstance() {
-        IssueService issueService = GithubIssueService.newInstance();
+        GithubApiService apiService = GithubCachingServiceContainer.getGithubService();
```
Comment (takecare) [inline batch-aws-lambda/src/main/java/com/novoda/github/reports/lambda/issue/IssuesServiceClient.java]: this should be `GithubServiceContainer.getGithubService()`

File: batch-aws-lambda/src/main/java/com/novoda/github/reports/lambda/issue/IssuesServiceClient.java [hunk 3]
```diff
@@ -6,55 +6,53 @@
...
-        return new IssuesServiceClient(issueService, dateConverter, responseRepositoryIssuePersistTransformer);
+        return new IssuesServiceClient(apiService, dateConverter, responseRepositoryIssuePersistTransformer);
     }
 
-    public static IssuesServiceClient newInstance(GithubApiService githubApiService,
-                                                  DatabaseCredentialsReader databaseCredentialsReader) {
-
-        IssueService issueService = GithubIssueService.newInstance(githubApiService);
+    public static IssuesServiceClient newInstance(DatabaseCredentialsReader databaseCredentialsReader) {
+        GithubApiService apiService = GithubCachingServiceContainer.getGithubService();
```
Comment (takecare) [inline batch-aws-lambda/src/main/java/com/novoda/github/reports/lambda/issue/IssuesServiceClient.java]: this should be `GithubServiceContainer.getGithubService()`

File: batch-aws-lambda/src/main/java/com/novoda/github/reports/lambda/pullrequest/ReviewCommentsServiceClient.java [hunk 4]
```diff
@@ -14,52 +14,52 @@
...
     private static final int DEFAULT_PER_PAGE_COUNT = 100;
 
-    private final PullRequestService pullRequestService;
+    private final GithubApiService apiService;
     private final DateToISO8601Converter dateConverter;
     private final ResponsePersistTransformer<RepositoryIssueEvent> responseRepositoryIssueEventPersistTransformer;
 
     public static ReviewCommentsServiceClient newInstance() {
-        PullRequestService pullRequestService = GithubPullRequestService.newInstance();
+        GithubApiService apiService = GithubCachingServiceContainer.getGithubService();
```
Comment (takecare) [inline batch-aws-lambda/src/main/java/com/novoda/github/reports/lambda/pullrequest/ReviewCommentsServiceClient.java]: this should be `GithubServiceContainer.getGithubService()`

**Comment by blundell** [comment]
yeah seems simpler ... for now :D

**Pull request merged**